### PR TITLE
fix(settings): fix audio 

### LINKED ISSUE #1405: bug(settings): output device is set to zero automatically
image

to reproduce, drag the slider to a non-zero value, navigate away from the settings page, then return.

### PR DESCRIPTION
**Pull request by sdwoodbury**
### What this PR does 📖

image

- the sliders are now ~~disabled~~ commented out  because that stuff isn't implemented. 
- the options to change the audio sampling rate and level of echo cancellation are removed because they were not implemented. 
- there is now a switch to toggle echo cancellation (echo cancellation is binary, there aren't any levels for it). 
- calls blink less regarding the audio config - blink provides a struct that can poll the hardware without further interaction with blink. This is now used in the coroutine in the audio settings page.  

### Which issue(s) this PR fixes 🔨

- Resolve #1405

### CODE REVIEW

**Comment by luisecm** [comment]
The automated tests will fail on this pr since there are tests asserting for texts displayed on Settings Audio - Sample Rate and Noise Suppression. So, there is no need to wait on the CI for approving this PR. I will update the tests as soon as this one is merged.

**Review by stavares843 — COMMENTED** [review]
in my opinion the sliders being there but deactivated seems a bug

i believe we should add a coming soon when hovering instead

**Comment by sdwoodbury** [comment]
> in my opinion the sliders being there but deactivated seems a bug
> 
> i believe we should add a coming soon when hovering instead

or we could just comment them out

**Comment by stavares843** [comment]
or that lol

File: common/src/state/configuration.rs [hunk 1]
```diff
@@ -72,7 +72,7 @@ pub struct AudioVideo {
 impl Default for AudioVideo {
     fn default() -> Self {
         Self {
-            noise_suppression: false,
+            echo_cancellation: false,
```
Comment (lgmarchi) [inline common/src/state/configuration.rs]: Make sense serde default be true, and impl Default be false?

**Review by stavares843 — APPROVED** [review]

File: ui/src/components/settings/sub_pages/audio.rs [hunk 2]
```diff
@@ -150,17 +172,18 @@ pub fn AudioSettings(cx: Scope) -> Element {
...
-                    aria_label: "range-input-device".into(),
-                    initial_value: 100.0,
-                    min: VOL_MIN,
-                    max: VOL_MAX,
-                    icon_left: Shape::Microphone,
-                    icon_right: Shape::MicrophoneWave,
-                    onchange: move |_| {}
-                }
-            }
+            // SettingSectionSimple {
```
Comment (lgmarchi) [inline ui/src/components/settings/sub_pages/audio.rs:175]: Can we add TODO for why we let it commented?

File: ui/src/components/settings/sub_pages/audio.rs [hunk 3]
```diff
@@ -169,57 +192,64 @@ pub fn AudioSettings(cx: Scope) -> Element {
...
-                    aria_label: "range-output-device".into(),
-                    initial_value: VOL_MIN,
-                    min: 0.0,
-                    max: VOL_MAX,
-                    icon_left: Shape::Speaker,
-                    icon_right: Shape::SpeakerWave,
-                    onchange: move |_| {}
-                }
-            }
+            // SettingSectionSimple {
```
Comment (lgmarchi) [inline ui/src/components/settings/sub_pages/audio.rs:199]: Same here about TODO.

File: ui/src/components/settings/sub_pages/audio.rs [hunk 4]
```diff
@@ -1,28 +1,26 @@
...
 use warp::logging::tracing::log;
 
-use crate::components::settings::{SettingSection, SettingSectionSimple};
+use crate::components::settings::SettingSection;
 use common::state::{action::ConfigAction, Action, State};
 use common::{sounds, WARP_CMD_CH};
 
-pub const VOL_MIN: f32 = 0.0;
-pub const VOL_MAX: f32 = 200.0;
+// pub const VOL_MIN: f32 = 0.0;
```
Comment (lgmarchi) [inline ui/src/components/settings/sub_pages/audio.rs:15]: Add a TODO why we let it commented

**Review by lgmarchi — APPROVED** [review]
Just left a question and suggetions

**Review by dariusc93 — APPROVED** [review]

File: common/src/state/configuration.rs (hunk 1 above)
Comment (sdwoodbury) [reply common/src/state/configuration.rs]: "does this make sense?" no. i think that if  a struct implements Default, it shouldn't have `#[serde(default)]` on any of its attributes.

**Pull request merged**